composer/utils/checkpoint.py

```diff
@@ -518,21 +510,35 @@ def read_data(self, plan: LoadPlan, planner: LoadPlanner):
                 # Ensure state exists
                 state_dict['state'] = state_dict.get('state', {})
 
+            # Only some ranks are meant to load checkpoint
+            expect_file = False
+            process_group = None
+            device_mesh = state.fsdp_device_mesh
+            if device_mesh is not None and device_mesh.ndim == 2:
```

Comment:
Little comment addition here
```suggestion
            # device_mesh.ndim == 2 signifies hybrid sharding.
            if device_mesh is not None and device_mesh.ndim == 2:
```